.github/scripts/ci_failure_bot.py

```
@@ -0,0 +1,170 @@
+            for job in jobs:
+                if job.conclusion == "failure":
+                    for step in job.steps:
+                        if step.conclusion == "failure":
+                            failed_jobs.append(
+                                {
+                                    "job_name": job.name,
+                                    "step_name": step.name,
+                                    "step_number": step.number,
+                                }
+                            )
+            return failed_jobs
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)